Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12AuditLogger.java

```diff
@@ -0,0 +1,216 @@
+ * DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES 
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; 
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND 
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT 
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS 
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+ */
+package gov.hhs.fha.nhinc.corex12.docsubmission.audit;
+
+import gov.hhs.fha.nhinc.auditrepository.nhinc.proxy.AuditRepositoryProxy;
+import gov.hhs.fha.nhinc.auditrepository.nhinc.proxy.AuditRepositoryProxyObjectFactory;
+import gov.hhs.fha.nhinc.common.auditlog.LogEventRequestType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AcknowledgementType;
+import gov.hhs.fha.nhinc.common.nhinccommon.AssertionType;
+import gov.hhs.fha.nhinc.common.nhinccommon.NhinTargetSystemType;
+import gov.hhs.fha.nhinc.nhinclib.NhincConstants;
+import org.apache.log4j.Logger;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmission;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeBatchSubmissionResponse;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeRequest;
+import org.caqh.soap.wsdl.corerule2_2_0.COREEnvelopeRealTimeResponse;
+
+/**
+ * Class provides APIs to persist Audit data
+ * 
+ * @author svalluripalli
+ */
+public class COREX12AuditLogger {
+
+    private static final Logger LOG = Logger.getLogger(COREX12AuditLogger.class);
+    private final COREX12RealtimeTransforms x12AuditTransformer = new COREX12RealtimeTransforms();
+    private final COREX12BatchAuditTransforms x12BatchAuditTransformer = new COREX12BatchAuditTransforms();
+
+
+    /**
+     *
+     * @param message
+     * @param assertion
+     * @param target
+     * @param direction
+     */
+    public void auditNhinCoreX12RealtimeRequest(COREEnvelopeRealTimeRequest message, AssertionType assertion, NhinTargetSystemType target,
+        String direction, boolean isRequesting) {
+        LOG.trace("---Begin COREX12AuditLogger.auditCoreX12RealtimeRequest()---");
+        // Set up the audit logging request message
+        LogEventRequestType auditLogMsg = logNhinCoreX12RealtimeRequest(message, assertion, target, direction, isRequesting);
+        if (auditLogMsg != null && auditLogMsg.getAuditMessage() != null) {
+            audit(auditLogMsg, assertion);
+        } else {
+            LOG.error("Core X12 Realtime Request auditLogMsg is null");
+        }
+        LOG.trace("---End COREX12AuditLogger.auditCoreX12RealtimeRequest()---");
+    }
+
+    /**
+     *
+     * @param message
+     * @param assertion
+     * @param target
+     * @param direction
+     * @param isRequesting
+     */
+    public void auditNhinCoreX12RealtimeRespponse(COREEnvelopeRealTimeResponse message, AssertionType assertion, NhinTargetSystemType target,
```

Comment:
Previous naming conventions also apply to function names (e.g., we use setInboundCORE_X12DSRealTime(...) instead of setInboundCoreX12DSRealTime(...))